root/programs/C#/Frameworks/Infrastructure/Public/Util/GetHash.cs

```diff
@@ -82,40 +88,64 @@ public enum EnumHashAlgorithm
         SHA512Managed,
     };
 
+    #endregion
+
+    #region GetHash
+
     /// <summary>ハッシュを取得するクラス</summary>
     public class GetHash
     {
+        #region GetSaltedPassword
+
+        /// <summary>パスワードをDB保存する際には塩味パスワードとして保存する。</summary>
+        /// <param name="rawPassword">ユーザが入力した生のパスワード</param>
+        /// <param name="eha">ハッシュ・アルゴリズム列挙型</param>
+        /// <param name="saltLength">ソルトの文字列長</param>
+        /// <returns>塩味パスワード</returns>
+        /// <see ref="http://www.atmarkit.co.jp/ait/articles/1110/06/news154_2.html"/>
+        public static string GetSaltedPassword(string rawPassword, EnumHashAlgorithm eha, int saltLength)
+        {
+            // overlordへ
+            return GetHash.GetSaltedPassword(rawPassword, eha, saltLength, 1);
+        }
+
         /// <summary>パスワードをDB保存する際には塩味パスワードとして保存する。</summary>
-        /// <param name="rawPasswd">ユーザが入力した生のパスワード</param>
+        /// <param name="rawPassword">ユーザが入力した生のパスワード</param>
         /// <param name="eha">ハッシュ・アルゴリズム列挙型</param>
         /// <param name="saltLength">ソルトの文字列長</param>
+        /// <param name="stretchCount">ストレッチ回数</param>
         /// <returns>塩味パスワード</returns>
-        public static string GetSaltedPasswd(string rawPasswd, EnumHashAlgorithm eha, int saltLength)
+        public static string GetSaltedPassword(string rawPassword, EnumHashAlgorithm eha, int saltLength, int stretchCount)
         {
             // ランダム・ソルト文字列を生成（区切り記号は含まなくても良い）
-            string salt = Membership.GeneratePassword(saltLength, 0);
-            // 塩味パスワード（文字列）を生成して返す。
-            return salt + GetHash.GetHashString(salt + rawPasswd, eha);
+            string salt = GetPassword.Generate(saltLength, 0); //Membership.GeneratePassword(saltLength, 0);
 
+            // 塩味パスワード（文字列）を生成して返す。
+            return
+                CustomEncode.ToBase64String(CustomEncode.StringToByte(salt, CustomEncode.UTF_8))
+                + "." + CustomEncode.ToBase64String(CustomEncode.StringToByte(stretchCount.ToString(), CustomEncode.UTF_8))
+                + "." + CustomEncode.ToBase64String(CustomEncode.StringToByte(GetHash.GetHashString(salt + rawPassword, eha, stretchCount), CustomEncode.UTF_8));
             // バイト配列仕様は、フィールドが文字列の可能性が高いので辞めた。
         }
 
         /// <summary>パスワードを比較して認証する。</summary>
-        /// <param name="rawPasswd">ユーザが入力した生のパスワード</param>
-        /// <param name="saltedPasswd">塩味パスワード</param>
+        /// <param name="rawPassword">ユーザが入力した生のパスワード</param>
+        /// <param name="saltedPassword">塩味パスワード</param>
```

Comment:
@daisukenishino 

Could you revice the related code?
(e.g. [this code](https://github.com/OpenTouryoProject/OpenTouryo/pull/206/files#diff-cb6cd6992edec91f48dcce4b6a451444R107))